vllm/config.py

```diff
@@ -4933,3 +4933,42 @@ def get_layers_from_vllm_config(vllm_config: VllmConfig,
         vllm_config.compilation_config.static_forward_context.items()
         if isinstance(layer, layer_type)
     }
+
+
+@config
+@dataclass
+class SpeechToTextConfig:
+    """Configuration for speech-to-text models."""
+
+    sample_rate: float = 16_000
+    """Sample rate (Hz) to resample input audio to. Most speech models expect
+    16kHz audio input. The input audio will be automatically resampled to this
+    rate before processing."""
+
+    hop_length: Optional[int] = None
+    """Number of samples between successive frames in the log-mel spectrogram.
+    If None, uses the model's default hop length. This affects the temporal
+    resolution of the audio features and is used for calculating prompt token
+    counts."""
+
+    max_audio_clip_s: int = 30
+    """Maximum duration in seconds for a single audio clip without chunking.
+    Audio longer than this will be split into smaller chunks if
+    `allow_audio_chunking` is enabled, otherwise it will be rejected."""
+
+    allow_audio_chunking: bool = False
+    """Whether to allow splitting long audio files into smaller chunks.
+    When True, audio longer than `max_audio_clip_s` will be automatically
+    split with overlapping segments. When False, long audio will be rejected.
+    """
```

Comment:
```suggestion
    max_audio_clip_s: Optional[int] = 30
    """Maximum duration in seconds for a single audio clip without chunking.
    Audio longer than this will be split into smaller chunks if
    `allow_audio_chunking` is enabled, otherwise it will be rejected."""
```

Think it's always nicer to try to remove such flags. Couldn't we get the same behavior of "allow_audio_chunking = False" by setting `max_audio_clip_s` to `None`?